### PR TITLE
feat(core): Add "watch asset" hook and implement corresponding tests

### PR DESCRIPTION
**Pull request by deeseeker**
Screenshot 2024-08-28 100943
Screenshot 2024-08-28 101502
I am getting this error after testing. Could you please review the code?

### CODE REVIEW

File: packages/core/tsconfig.vitest-temp.json
Comment (fracek) [inline packages/core/tsconfig.vitest-temp.json:1]: Please check what you commit before pushing.

**Comment by fracek** [comment]
Thank you for this. Can you make sure the tests pass before pushing?

**Review by jaipaljadeja — CHANGES_REQUESTED** [review]
The tests currently work with Mock Connector ([mock.ts](https://github.com/apibara/starknet-react/blob/main/packages/core/src/connectors/mock.ts)).
Please take note of the request method. Add a case for watchAsset, similar to `wallet_addStarknetChain` and ensure that it returns a boolean. Be sure to double-check what gets returned for this case referring the @starknet-io/types-js package.

Additionally, refer to the test for `useAddChain`, and rewrite the test accordingly.

**Review by fracek — APPROVED** [review]
Looks good now, thank you!

**Review by jaipaljadeja — APPROVED** [review]
Looks good to me!

**Comment by deeseeker** [comment]
> Looks good now, thank you!

It's an honor sir!

**Pull request merged**